pkg/apis/deployment/v1alpha/secret_hashes.go

```diff
@@ -34,11 +34,13 @@ type SecretHashes struct {
 	TLSCA string `json:"tls-ca,omitempty"`
 	// SyncTLSCA contains the hash of the sync.tls.caSecretName secret
 	SyncTLSCA string `json:"sync-tls-ca,omitempty"`
+	// User's map contains hashes for each user
+	Users map[string]string `json:"users,omitempty"`
 }
 
 // Equal compares two SecretHashes
 func (sh *SecretHashes) Equal(other *SecretHashes) bool {
-	if sh == nil || other == nil {
+	if other == nil {
```

Comment:
What if sh is nil? This check allows us to prevent NullPointerExceptions

Reply:
changed